### PR TITLE
packaging: add user and labels checks

### PR DESCRIPTION
**Pull request by graphaelli**
for https://github.com/elastic/apm-server/issues/1693

### CODE REVIEW

File: dev-tools/packaging/package_test.go [hunk 1]
```diff
@@ -382,6 +384,31 @@ func checkDockerEntryPoint(t *testing.T, p *packageFile, info *dockerInfo) {
...
+	}
+	if license, present := info.Config.Labels["license"]; !present || license != expectedLicense {
+		t.Errorf("unexpected license label: %s", license)
+	}
+}
+
+func checkDockerUser(t *testing.T, info *dockerInfo) {
+	if info.Config.User == "root" {
+		t.Errorf("unexpected docker user: %s", info.Config.User)
+	}
```
Comment (jsoriano) [inline dev-tools/packaging/package_test.go]: I think auditbeat uses root.

File: dev-tools/packaging/package_test.go [hunk 2]
```diff
@@ -382,6 +384,31 @@ func checkDockerEntryPoint(t *testing.T, p *packageFile, info *dockerInfo) {
 	})
 }
 
+func checkDockerLabels(t *testing.T, info *dockerInfo, file string) {
+	if vendor := info.Config.Labels["org.label-schema.vendor"]; vendor != "Elastic" {
+		t.Errorf("unexpected vendor label: %s", vendor)
+	}
```
Comment (jsoriano) [inline dev-tools/packaging/package_test.go]: Should we provide a way to skip Elastic-specific tests for community beats?

File: dev-tools/packaging/package_test.go (hunk 1 above)
Comment (graphaelli) [reply dev-tools/packaging/package_test.go]: Bummer, it does, I guess package tests aren't run on pull requests?  I'll look for a workaround for this.

Comment (graphaelli) [reply dev-tools/packaging/package_test.go]: counting on the fact that root ownership means run as root

File: dev-tools/packaging/package_test.go (hunk 2 above)
Comment (jsoriano) [reply dev-tools/packaging/package_test.go]: @graphaelli wdyt about this?

Comment (graphaelli) [reply dev-tools/packaging/package_test.go]: I agree we should provide a mechanism for this, I just hadn't come up with a good way short of wiring through a command line flag for it.  I've pushed up a commit to only perform this check if the vendor label is `Elastic`, which the user should customize.  This of course relies on that label working so it's not my favorite solution but should be good enough to close this out finally - ok?

Comment (jsoriano) [reply dev-tools/packaging/package_test.go]: :+1:

**Comment by jsoriano** [comment]
LGTM, @kvch could you also review this? Thanks!

**Pull request merged**